sdk/go/common/resource/asset.go

```diff
@@ -603,42 +612,43 @@ func DeserializeArchive(obj map[string]interface{}) (*Archive, bool, error) {
 		hash = h
 	}
 
-	var assets map[string]interface{}
-	if v, has := obj[ArchiveAssetsProperty]; has {
-		assets = make(map[string]interface{})
-		if v != nil {
-			m, ok := v.(map[string]interface{})
-			if !ok {
-				return &Archive{}, false, fmt.Errorf("unexpected archive contents of type %T", v)
-			}
+	assets := make(map[string]interface{})
```

Comment:
Not sure this is safe? Looks like we could create an Archive with assets and path set?

Reply:
I updated the `DeserializeArchive` method such that it is not possible to construct conflicting fields:
 - When we check for the `path` property and it exists, we immediately return `NewPathArchive`
 - When we check for the `uri` property and it exists, we immediately return `NewURIArchive`
 - Finally if we didn't find `path` nor `uri`, we get the `assets` and default to an empty map 

At the start of the function, we check that the input object _must_ have one of `path`, `uri` or `assets`